src/Plugin/Field/FieldFormatter/ConsoleLinkFormatter.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\media_mpx\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Url;
+use Drupal\media\Entity\Media;
+use Drupal\media_mpx\Entity\Account;
+use Drupal\media_mpx\Plugin\media\Source\Media as MpxMediaSource;
+
+/**
+ * Field formatter to link to the mpx console.
+ *
+ * @FieldFormatter(
+ *   id = "media_mpx_console_link",
+ *   label = @Translation("mpx Console Link"),
+ *   description = @Translation("Suitable for use on fields storing the mpx guid value, the mpx console link formatter will create a link to the media in the mpx console."),
+ *   field_types = {
+ *     "string"
+ *   }
+ * )
+ */
+class ConsoleLinkFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $element = [];
+
+    if (!$this->appliesTo($items)) {
+      return $element;
+    }
+
+    /** @var \Drupal\media_mpx\Entity\Account $account */
+    $account = $items->getEntity()->getSource()->getAccount();
+
+    foreach ($items as $delta => $item) {
+      $element[$delta] = [
+        '#type' => 'link',
+        '#title' => $item->getString(),
+        '#url' => $this->getUrl($item, $account),
+      ];
+    }
+
+    return $element;
+  }
+
+  /**
+   * Does this field formatter apply to the given field items.
+   *
+   * @param \Drupal\Core\Field\FieldItemListInterface $items
+   *   The field values to be rendered.
+   *
+   * @return bool
+   *   TRUE if this field formatter applies, otherwise FALSE.
+   */
+  protected function appliesTo(FieldItemListInterface $items) {
+    if (!($entity = $items->getEntity()) || !$entity instanceof Media) {
+      return FALSE;
+    }
+
+    $source_plugin = $entity->getSource();
```

Comment:
Nitpick: It'd be great to add @var 
```
/* @var \Drupal\media_mpx\Plugin\media\Source\Media $source_plugin **/
```

Reply:
It's not really needed. Since the next line is a conditional checking the class, PHPStorm will infer the type beyond that point in the code and still give you autocomplete. I'm going to skip this if it's alright, since its'a bit of a pain to update all three relevant PR's in play.

Reply:
sounds good